cmd/pkl-gen-go/pkl-gen-go.go

```diff
@@ -146,6 +157,52 @@ func generatorSettingsSource() *pkl.ModuleSource {
 	return pkl.UriSource(fmt.Sprintf("package://pkg.pkl-lang.org/pkl-go/pkl.golang@%s#/GeneratorSettings.pkl", Version))
 }
 
+// mimick logic for finding project dir in the pkl CLI.
+func doFindProjectDir(dir string) string {
+	if fileExists(filepath.Join(dir, "PklProject")) {
+		return dir
+	}
+	parent := filepath.Dir(dir)
+	if parent == dir {
+		return ""
+	}
+	return doFindProjectDir(parent)
+}
+
+func findProjectDir(projectDirFlag string) string {
+	if projectDirFlag != "" {
+		return projectDirFlag
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		return ""
+	}
+	return doFindProjectDir(cwd)
```

Comment:
nit: Isn't it more idiomatic to write a loop instead of recursion in Go?
Also why return an empty string to indicate "not found" instead of `nil`?

Reply:
Empty string is the zero value; it seems a little more idiomatic to write Go in this way.

As for recursion vs. loop: I don't feel that one is more idiomatic over the other; given that this already works and this isn't going to be a perf bottleneck, I think I'll keep it as is.